q2_diversity_lib/tests/test_alpha.py

```diff
@@ -15,23 +15,28 @@
 from q2_types.feature_table import BIOMV210Format
 from q2_types.tree import NewickFormat
 from q2_diversity_lib import (faith_pd, pielou_evenness, observed_features,
-                              shannon_entropy)
+                              shannon_entropy, alpha, translate_metric_name)
+from qiime2 import Artifact
 
-nonphylogenetic_measures = [observed_features, pielou_evenness,
-                            shannon_entropy]
+nonphylogenetic_metrics = alpha.METRICS['NONPHYLO']['IMPL']
+metric_name_translations = alpha.METRICS['METRIC_NAME_TRANSLATIONS']
 
 
 class SmokeTests(TestPluginBase):
     package = 'q2_diversity_lib.tests'
 
     def setUp(self):
         super().setUp()
-        self.empty_table = biom.Table(np.array([]), [], [])
+        empty_table = biom.Table(np.array([]), [], [])
+        self.empty_table = Artifact.import_data('FeatureTable[Frequency]',
+                                                empty_table)
 
     def test_non_phylogenetic_passed_empty_table(self):
-        for measure in nonphylogenetic_measures:
+        for metric in nonphylogenetic_metrics:
+            metric_tr = translate_metric_name(metric, metric_name_translations)
```

Comment:
This doesn't seem worth the extra indirection - why not just keep all of the "logic" in one place - the `METRICS` data structure defined in the module?

Reply:
Am I right that it's the `metric_name_translations` variable you don't like here, not the `translate_metric_names` function?

I'm good with swapping that out, but think the function helps readability a lot.

Reply:
Oooh! Looks like it was `translate_metric_names()` after all? https://github.com/qiime2/q2-diversity/pull/281#discussion_r471161680

Reply:
Thanks for checking - its the function I'm concerned with - I think having the metrics "automatically" initialized if not listed in the dict might cause some headaches later. Adding a new metric, for example, what if I misspell it? This util will "allow" the misspelling to leak through. Not the end of the world, but then you're stuck debugging with some indirection in the way.

I'm also thinking about the _Zen of Python_ here: https://www.python.org/dev/peps/pep-0020/

> Explicit is better than implicit.
Simple is better than complex.

Reply:
Resolved in 56b88f3.

Reply:
Sorry, just noticed that you edited this before I had responded - somehow I had a stale version of https://github.com/qiime2/q2-diversity-lib/pull/24#discussion_r471812735.

> How do you feel about this?

I like that plan! I agree, the unimplemented metrics don't need to be in the dict.

Looks like 56b88f3 goes with the full list of unimpl (if I am reading right), so your call on the programmatic unimpl changes. Again, real sorry about the stale comment. That's one downside to retroactive comment editing - no notifications!

Reply:
resolved in https://github.com/qiime2/q2-diversity/pull/281/commits/a544885a85510aa04a0f3efeeebe500ccdf6291d and 5a6a620f070cbf02cecab6ef5ef3fc58043d2e3c